### PR TITLE
feat: [ID-1585] get signers after confirmation popup

### PR DESCRIPTION
**Pull request by scastleman-immutable**
# Summary
[Jira ticket](https://immutable.atlassian.net/browse/ID-1585)
Moved initialisation of starkex signers after the confirmation popup is initialised, in order to avoid possible popup blocking issues

### CODE REVIEW

File: packages/passport/sdk/src/starkEx/workflows/workflows.test.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    it(`should return an error for ${methodName}`, async () => {
+      expect(async () => await passportImxProvider[methodName!](args))
+        .rejects
+        .toThrow(
+          new PassportError(
+            'User has been logged out',
+            PassportErrorType.NOT_LOGGED_IN_ERROR,
+          ),
+        );
+    });
```
Comment (imx-mikhala) [inline packages/passport/sdk/src/starkEx/workflows/workflows.test.ts:77]: When asserting errors with custom types, like here with `PassportErrorType.NOT_LOGGED_IN_ERROR` jest doesn't actually assert this type is what we expect and only validates the message is the same. You could pass in any `PassportErrorType` here and the test will pass.

E.g. the following will pass the test

```
expect(async () => await passportImxProvider[methodName!](args))
        .rejects
        .toThrow(
          new PassportError(
            'User has been logged out',
            'Some type' as PassportErrorType,
          ),
        );
```

If you want to assert the type the only way I've found works is using a catch block to assert the type is what is expected. Note that we need to assert outside of the catch, otherwise we cant be sure that the catch was evaluated as the test will pass if the function doesn't throw.

```
let message = '';
let type = '';
try {
        await passportImxProvider[methodName!](args);
} catch (e: any) {
        message = e.message;
        type = e.type;
}

expect(message).toEqual('User has been logged out');
expect(type).toEqual(PassportErrorType.NOT_LOGGED_IN_ERROR);
```

Comment (scastleman-immutable) [reply packages/passport/sdk/src/starkEx/workflows/workflows.test.ts:77]: Thanks for the callout. I just moved this test over from `passportImxProvider.test.ts`, but probably a good opportunity to fix it up while we're in the area.

File: packages/passport/sdk/src/starkEx/workflows/workflows.test.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      passportEventEmitter.emit(PassportEvents.LOGGED_OUT);
+    });
+
+    it(`should return an error for ${methodName}`, async () => {
+      try {
+        await passportImxProvider[methodName!](args);
+      } catch (err: any) {
+        const { message, type } = err;
+        expect(message).toEqual('User has been logged out');
+        expect(type).toEqual(WorkflowErrorMap[methodName]);
```
Comment (imx-mikhala) [inline packages/passport/sdk/src/starkEx/workflows/workflows.test.ts:86]: Can we move the expect outside of the catch to ensure we are actually falling into the catch block? Otherwise if the function isnt actually throwing this test still passes. See my other comment for how to set this up.

**Review by oliyoung — APPROVED** [review]

File: packages/passport/sdk-sample-app/src/context/ImmutableProvider.tsx [hunk 3]
```diff
@@ -126,7 +126,7 @@ const getPassportConfig = (environment: EnvironmentNames): PassportModuleConfigu
           authenticationDomain: 'https://auth.dev.immutable.com',
           magicPublishableApiKey: 'pk_live_4058236363130CA9',
           magicProviderId: 'C9odf7hU4EQ5EufcfgYfcBaT5V6LhocXyiPRhIjw2EY=',
-          passportDomain: 'https://passport.dev.immutable.com',
```
Comment (haydenfowler) [inline packages/passport/sdk-sample-app/src/context/ImmutableProvider.tsx:129]: Was this change intentional?

**Review by haydenfowler — CHANGES_REQUESTED** [review]
The change to `ImmutableProvider.tsx` causes issues

image
